src/main/java/org/lwjglx/opengl/ContextAttribs.java

```diff
@@ -15,77 +15,440 @@
  */
 package org.lwjglx.opengl;
 
+import java.nio.IntBuffer;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.lwjglx.BufferUtils;
+import org.lwjglx.LWJGLUtil;
+
 public final class ContextAttribs {
 
-    public ContextAttribs() {}
+    // ATTRIBUTES
+
+    public static final int CONTEXT_MAJOR_VERSION_ARB = 0x2091;
+    public static final int CONTEXT_MINOR_VERSION_ARB = 0x2092;
+
+    public static final int CONTEXT_PROFILE_MASK_ARB = 0x9126, CONTEXT_CORE_PROFILE_BIT_ARB = 0x00000001,
+        CONTEXT_COMPATIBILITY_PROFILE_BIT_ARB = 0x00000002, CONTEXT_ES2_PROFILE_BIT_EXT = 0x00000004;
+
+    public static final int CONTEXT_FLAGS_ARB = 0x2094, CONTEXT_DEBUG_BIT_ARB = 0x0001,
+        CONTEXT_FORWARD_COMPATIBLE_BIT_ARB = 0x0002, CONTEXT_ROBUST_ACCESS_BIT_ARB = 0x00000004,
+        CONTEXT_RESET_ISOLATION_BIT_ARB = 0x00000008;
+
+    public static final int CONTEXT_RESET_NOTIFICATION_STRATEGY_ARB = 0x8256, NO_RESET_NOTIFICATION_ARB = 0x8261,
+        LOSE_CONTEXT_ON_RESET_ARB = 0x8252;
+
+    public static final int CONTEXT_RELEASE_BEHABIOR_ARB = 0x2097, CONTEXT_RELEASE_BEHAVIOR_NONE_ARB = 0x0000,
+        CONTEXT_RELEASE_BEHAVIOR_FLUSH_ARB = 0x2098;
+
+    public static final int CONTEXT_LAYER_PLANE_ARB = 0x2093; // WGL-only
+
+    // STATE
+
+    private int majorVersion;
+    private int minorVersion;
+
+    private int profileMask;
+    private int contextFlags;
+
+    private int contextResetNotificationStrategy = NO_RESET_NOTIFICATION_ARB;
+    private int contextReleaseBehavior = CONTEXT_RELEASE_BEHAVIOR_FLUSH_ARB;
+
+    private int layerPlane;
 
-    public ContextAttribs(final int majorVersion, final int minorVersion) {}
+    // CONSTRUCTORS
+
+    /**
+     * Creates the default ContextAttribs instance. No special attributes will be used when creating the OpenGL context.
+     */
+    public ContextAttribs() {
+        this(1, 0);
+    }
+
+    /** Creates a ContextAttribs instance for the given OpenGL version. */
+    public ContextAttribs(int majorVersion, int minorVersion) {
+        this(majorVersion, minorVersion, 0, 0);
+    }
+
+    /**
+     * Creates a new ContextAttribs instance with the given attributes.
+     *
+     * @param majorVersion the major OpenGL version
+     * @param minorVersion the minor OpenGL version
+     * @param profileMask  the context profile mask. One of:<br>
+     *                     {@link #CONTEXT_CORE_PROFILE_BIT_ARB}, {@link #CONTEXT_FORWARD_COMPATIBLE_BIT_ARB},
+     *                     {@link #CONTEXT_ES2_PROFILE_BIT_EXT}
+     */
+    public ContextAttribs(int majorVersion, int minorVersion, int profileMask) {
+        this(majorVersion, minorVersion, 0, profileMask);
```

Comment:
`profileMask` should be third parameter, shouldn't it?

Reply:
This is copied directly from LWJGL2 sources, so if it was buggy there, I have to keep the same behaviour.